.github/workflows/test.yml

```diff
@@ -1,14 +1,53 @@
-name: "test"
+name: "integration tests"
 on:
   pull_request:
     paths-ignore:
       - '**.md'
   push:
     branches:
       - master
+env:
+  # Variables defined in the repository
+  SENTRY_ORG: ${{ vars.SENTRY_ORG }}
+  # For master we have an environment variable that selects the action-release project
+  # instead of action-release-prs
+  # For other branches: https://sentry-ecosystem.sentry.io/releases/?project=4505075304693760
+  # For master branch: https://sentry-ecosystem.sentry.io/releases/?project=6576594
+  SENTRY_PROJECT: ${{ vars.SENTRY_PROJECT }}
 
 jobs:
-  mock-release: # Make sure that the action works on a clean machine without building
+  # You're welcome to make changes on this job as part of your PR in order to test out your changes
+  # We can always undo the changes once we're satisfied with the results
+  #
+  # Secrets on this repo do not get shared with PRs openeed on a fork, thus,
+  # Add SENTRY_AUTH_TOKEN as a secret to your fork if you want to use this job.
+  # Checkout the README.md in how to create the internal integration (read: auth token)
+  create-real-release-per-push:
+    name: "Test current action"
+    runs-on: ubuntu-latest
+    # XXX: This job will fail for forks, skip step on forks and let contributors tweak it when ready
+    if: github.ref != 'refs/heads/master'
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        # For PRs, this supports creating a release using the commits from the branch (rather than the merge commit)
+        ref: ${{ github.event.pull_request.head.sha || github.sha }}
+
+    # This allows executing the action's code in the next step rather than a specific tag
+    - uses: './.github/actions/use-local-dockerfile'
```

Comment:
This is basically an sed command to tell the action to use the `Dockerfile` found in the checkout.

```diff
diff --git a/action.yml b/action.yml
index 89bbe57..4095cbe 100644
--- a/action.yml
+++ b/action.yml
@@ -47,7 +47,7 @@ inputs:
 runs:
   using: 'docker'
   # If you change this, update the use-local-dockerfile action
-  image: 'docker://ghcr.io/getsentry/action-release-image:latest'
+  image: 'Dockerfile'
 branding:
   icon: 'triangle'
   color: 'purple'
```